src/sources/postgresql_metrics.rs

```diff
@@ -505,11 +540,7 @@ impl PostgresqlMetrics {
                     tags!(self.tags, "db" => db),
                 ),
             ]);
-            if row
-                .columns()
-                .iter()
-                .any(|column| column.name() == "checksum_failures")
-            {
+            if self.version.unwrap() >= 120000 {
```

Comment:
I think this is another case we should clean-up similar to the `Option<Client>` we have. I think `self.version` should ideally just be a `usize`.

This particular one could be cleaned up by fetching the version every collection. I think this would be preferable anyway as the version could change after `vector` is started if they upgrade the database in-place.

I like that you are verifying the version now though! I think that'll be useful for users.

Reply:
Yes, absolutely, going to address this with `Option<Client>` (mentioned this in PR description)